### PR TITLE
RPCh SDK part 1/3

### LINKED ISSUE #32: implementation of RPCh SDK 1/3
### Description

Implement part 1/3 of RPCh SDK.

### Relevant issues

- [x] retrofit hoprd component from [RPCh prototype](https://github.com/hoprnet/RPCh)
- [x] retrofit manager component to request-cache component from [RPCh prototype](https://github.com/hoprnet/RPCh)
- [ ] add testing, coverage >90%
- [ ] add documentation

### Definition of DONE

Once we have a working RPCh SDK library with limited capabilities.

### PR DESCRIPTION
**Pull request by diegoalzate**
Closes #32

### Tasks
- [x] retrofit hoprd component from [RPCh prototype](https://github.com/hoprnet/RPCh)
- [x] retrofit manager component to request-cache component from [RPCh prototype](https://github.com/hoprnet/RPCh)
- [x] add testing, coverage > 90%
- [x] add documentation

### CODE REVIEW

File: packages/sdk/src/index.spec.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import assert from "assert";
+import SDK from "./index";
+import {
+  PEER_ID_A as ORIGIN,
+  PROVIDER,
+  RPC_REQ_SMALL,
+} from "rpch-commons/src/fixtures";
```
Comment (nionis) [inline packages/sdk/src/index.spec.ts:7]: it doesn't look like fixtures are exported yet, so before we merge this PR we need to make sure `rpch/commons` is updated to export fixtures as this is not the right way to important them

**Review by nionis — COMMENTED** [review]
one minor comment, progress looks good 👍

File: packages/sdk/src/index.spec.ts (hunk 1 above)
Comment (diegoalzate) [reply packages/sdk/src/index.spec.ts:8]: added export to commons 👍

**Review by nionis — CHANGES_REQUESTED** [review]
Amazing first PR! 💪

- small logic change
- some cosmetic changes

File: packages/sdk/src/request-cache.ts [hunk 2]
```diff
@@ -1,191 +1,92 @@
...
+    this.requests.delete(req.id);
+  }
 
-//     logVerbose("requests", this.requests.size);
-//     logVerbose("segments", this.segments.size);
+  /**
+   * Resolve request promise and delete the request from map
+   * @param res Response received from cache module
+   */
+  public onResponseFromSegments(res: Response): void {
```
[the remark this answers is not in the document]
Comment (diegoalzate) [reply packages/sdk/src/request-cache.ts:59]: I think that is a good suggestion. I feel like the request-cache was doing too much

File: packages/sdk/src/index.ts [hunk 3]
```diff
@@ -1,21 +1,74 @@
...
+        }
+      }
+    );
+    this.requestCache.setInterval();
+  }
+
+  public createRequest(
+    origin: string,
+    provider: string,
+    destination: string
```
Comment (nionis) [inline packages/sdk/src/index.ts:54]:
```suggestion
    body: string
```

File: packages/sdk/src/index.ts [hunk 4]
```diff
@@ -1,21 +1,74 @@
...
+  ): Request {
+    return Request.fromData(origin, provider, destination);
```
Comment (nionis) [inline packages/sdk/src/index.ts:56]:
```suggestion
    return Request.fromData(origin, provider, body);
```

**Review by nionis — APPROVED** [review]

**Pull request merged**